Source Indexing fails for Portable PDBs in the "Index Sources & Publish Symbols" step resulting in a partially succeeded build.

If you have a "Index Sources & Publish Symbols" step and you're building let's say an ASP.NET Core application (even on .NET and not necessarily on .NET Core) you're likely getting defaulted to Portable PDBs in the build step and are having this build step fail.

It doesn't result in a full build failure, but instead a partially succeeded build.

Below is an example of such a log entry

2017-06-13T06:08:32.7107849Z ##[section]Starting: Publish symbols path: 
2017-06-13T06:08:32.7107849Z ==============================================================================
2017-06-13T06:08:32.7107849Z Task         : Index Sources & Publish Symbols
2017-06-13T06:08:32.7107849Z Description  : Index your source code and publish symbols to a file share
2017-06-13T06:08:32.7107849Z Version      : 1.0.36
2017-06-13T06:08:32.7107849Z Author       : Microsoft Corporation
2017-06-13T06:08:32.7107849Z Help         : Index your source code and publish symbols to a file share<br/>[More Information](https://go.microsoft.com/fwlink/?LinkID=613722)
2017-06-13T06:08:32.7107849Z ==============================================================================
2017-06-13T06:08:34.2345798Z Found 1 files.
2017-06-13T06:08:34.8256000Z ##[error]Indexed source information could not be retrieved from 'd:\a\1\a\final\BPerfWebViewer\BPerfWebViewer.pdb'. Symbol indexes could not be retrieved.
2017-06-13T06:08:34.8486004Z ##[section]Finishing: Publish symbols path:

--- a/Tasks/PublishSymbols/IndexHelpers/IndexFunctions.ps1
+++ b/Tasks/PublishSymbols/IndexHelpers/IndexFunctions.ps1
@@ -51,6 +51,13 @@ function Invoke-IndexSources {
                 #    continue
                 #}
 
+                $bytes = Get-Content $symbolsFilePath -Encoding byte -TotalCount 4
+                $data = [System.Text.Encoding]::ASCII.GetString($bytes)
+                if ($data.equals("BSJB")) {
+                    Write-Verbose "Skipping: $symbolsFilePath because it is a Portable PDB"
+                    continue
+                }
+
                 # Get the source file paths embedded in the symbols file.
                 [string[]]$sourceFilePaths = Get-SourceFilePaths -SymbolsFilePath $symbolsFilePath -SourcesRootPath $provider.SourcesRootPath -TreatNotIndexedAsWarning:$TreatNotIndexedAsWarning
                 if (!$sourceFilePaths.Count) {

